addons/l10n_in/models/account.py

```diff
@@ -10,23 +10,7 @@ class AccountJournal(models.Model):
     _inherit = "account.journal"
 
     # Use for filter import and export type.
-    l10n_in_gstin_partner_id = fields.Many2one('res.partner', string="GSTIN Unit", ondelete="restrict", help="GSTIN related to this journal. If empty then consider as company GSTIN.")
-
-    def name_get(self):
-        """
-            Add GSTIN number in name as suffix so user can easily find the right journal.
-            Used super to ensure nothing is missed.
-        """
-        result = super().name_get()
-        result_dict = dict(result)
-        indian_journals = self.filtered(lambda j: j.company_id.account_fiscal_country_id.code == 'IN' and
-            j.l10n_in_gstin_partner_id and j.l10n_in_gstin_partner_id.vat)
-        for journal in indian_journals:
-            name = result_dict[journal.id]
-            name += "- %s" % (journal.l10n_in_gstin_partner_id.vat)
-            result_dict[journal.id] = name
-        return list(result_dict.items())
-
+    l10n_in_gstin_partner_id = fields.Many2one('res.partner', string="GSTIN Unit (Deprecated)", ondelete="restrict", help="GSTIN related to this journal. If empty then consider as company GSTIN.", deprecated=True)
```

Comment:
Why do you keep the field in deprecated True?  (why not remove it?)

Reply:
@jco-odoo If we directly remove it then we need to create a company and move all data to the newly created company.

And this is super complex to move all data to new companies that link to that unit to like invoices>sales->subscriptions. stock that related to sales etc.

Reply:
@jva-odoo But it is what needs to happen.  Complex or not.

Reply:
@jco-odoo Okay, can we just add deprecated and remove it in the next version?
so we have big-time to write complex migration.